tests/test_telemetry.py

```diff
@@ -369,3 +369,31 @@ def test_app_dependencies_loaded(self):
         for dependency, seen in seen_loaded_dependencies.items():
             if not seen:
                 raise Exception(dependency + " not recieved in app-dependencies-loaded message")
+
+    def setup_app_product_change(self):
```

Comment:
Can you do a dedicated scenario with appsec, profiling ... set to false at startup time.